### PR TITLE
Added is_obstructed? method

### PR DESCRIPTION
**Pull request by jonianven**

### CODE REVIEW

**Comment by yuzu-r** [comment]
Got some fails when pieces stand between the start and end points.

**Comment by jonianven** [comment]
Turns out ranges going down i.e. (3..1) -- were returning the range. Downto should be used instead. This stack overflow post explains: http://stackoverflow.com/questions/8822764/why-will-a-range-not-work-when-descending 

Also used upto to keep things consistent. All tests should pass now.

File: spec/helpers/games_helper_spec.rb [hunk 1]
```diff
@@ -11,5 +11,5 @@
 #   end
 # end
 RSpec.describe GamesHelper, type: :helper do
```
Comment (gruzzlymug) [inline spec/helpers/games_helper_spec.rb:13]: You might want to just delete these unused files if you don't have a clear use for them. They can always be re-added. I see you did that below 🌙

File: spec/models/piece_spec.rb [hunk 2]
```diff
@@ -1,5 +1,226 @@
 require 'rails_helper'
 
 RSpec.describe Piece, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  describe "is_obstructed? returns false if path is not blocked" do
```
Comment (gruzzlymug) [inline spec/models/piece_spec.rb:4]: very thorough testing. you may or may not need to every permutation. that's a judgement call.

I would encourage you to add a test where the move is invalid (not horiz, vert or diag). (1,1) to (4,3), for example. what does is_obstructed do then?

Other ideas
- move to same spot (1,1) to (1,1)
- move off the board (1,1) to (100, 3)

Comment (yuzu-r) [inline spec/models/piece_spec.rb:4]: We weren't really sure if checking for out of bounds, start  = destination, or move <> vertical/horizontal/diagonal should belong to a method called is_obstructed. if we start checking for vertical/horizontal/diagonal don't we have to also start coding to allow for the knight? or create a separate is_obstructed? for the knight class.

**Comment by jonianven** [comment]
@yuzu-r: I added checking for bounds. Start = destination should be checked in line 13 of piece.rb. Knight-like or invalid movements are not accounted for in is_obstructed with this commit (that will probably be specific to their class like you said... still not sure how we will approach that). I think you can go ahead & write tests for the two cases (bounds, start = end).

**Comment by kearlyaa** [comment]
We might consider adding a check in the is_obstructed? method for knights after we check the destination is not out of bounds and not occupied by a friendly piece.  I think we can just say if knight, return false after those other checks.

The rest of the PR looks good and I think you can safely merge or add the check for the knight and then merge.

**Comment by jonianven** [comment]
Okay, thanks @kearlyaa for the suggestion... I've added it in the latest commit 😄 @yuzu-r -- I am going to merge right now and you can add the tests later in a separate PR. I don't want to hold back the is_valid? tasks and I think we should be fine not testing those cases for the time being.

**Comment by yuzu-r** [comment]
Thanks for the update @jonianven. Additional tests are also passing and I will commit my additional changes in a separate PR.

**Pull request merged**